bigquery/src/main/java/org/apache/iceberg/gcp/bigquery/BigQueryMetastoreClientImpl.java

```diff
@@ -0,0 +1,594 @@
+        }
+      };
+
+  private final Bigquery client;
+  private final BigQueryOptions bigqueryOptions;
+
+  // Retry config with error messages and regex for rate limit exceeded errors.
+  private static final BigQueryRetryConfig DEFAULT_RETRY_CONFIG =
+      BigQueryRetryConfig.newBuilder()
+          .retryOnMessage(BigQueryErrorMessages.RATE_LIMIT_EXCEEDED_MSG)
+          .retryOnMessage(BigQueryErrorMessages.JOB_RATE_LIMIT_EXCEEDED_MSG)
+          .retryOnRegEx(BigQueryErrorMessages.RetryRegExPatterns.RATE_LIMIT_EXCEEDED_REGEX)
+          .build();
+
+  public static final ExceptionHandler BIGQUERY_EXCEPTION_HANDLER =
+      ExceptionHandler.newBuilder()
+          .abortOn(RuntimeException.class)
+          .retryOn(
+              java.net.ConnectException
+                  .class) // Retry on connection failures due to transient network issues.
+          .retryOn(
+              java.net.UnknownHostException
+                  .class) // Retry to recover from temporary DNS resolution failures.
+          .addInterceptors(EXCEPTION_HANDLER_INTERCEPTOR)
+          .build();
+
+  /** Constructs a client of the Google BigQuery service. */
+  public BigQueryMetastoreClientImpl(BigQueryOptions options)
+      throws IOException, GeneralSecurityException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests
+    HttpCredentialsAdapter httpCredentialsAdapter =
+        new HttpCredentialsAdapter(
+            GoogleCredentials.getApplicationDefault().createScoped(BigqueryScopes.all()));
+    this.client =
+        new Bigquery.Builder(
+                GoogleNetHttpTransport.newTrustedTransport(),
+                GsonFactory.getDefaultInstance(),
+                httpRequest -> {
+                  httpCredentialsAdapter.initialize(httpRequest);
+                  // Instead of throwing exceptions, analyze the HttpResponse object and inspect its
+                  // status code. This will allow BigQuery API errors to be converted into Iceberg
+                  // exceptions.
+                  httpRequest.setThrowExceptionOnExecuteError(false);
+                })
+            .setApplicationName("BigQuery Metastore Iceberg Catalog Plugin")
+            .build();
+    this.bigqueryOptions = options;
+  }
+
+  @Override
+  public Dataset create(Dataset dataset) {
+    Dataset response = null;
+    try {
+      response =
+          BigQueryRetryHelper.runWithRetries(
+              () -> internalCreate(dataset),
+              bigqueryOptions.getRetrySettings(),
+              BIGQUERY_EXCEPTION_HANDLER,
+              bigqueryOptions.getClock(),
+              DEFAULT_RETRY_CONFIG);
+    } catch (BigQueryRetryHelper.BigQueryRetryHelperException e) {
+      handleBigQueryRetryException(e);
+    }
+    return response;
+  }
+
+  @SuppressWarnings("FormatStringAnnotation")
```

Comment:
no need to suppress this as you can fix it by updating the exception block to `throw new AlreadyExistsException("Namespace already exists: %s", dataset.getId());`